tests/load/pipeline/test_arrow_loading.py

```diff
@@ -296,3 +300,57 @@ def some_data():
     # Load is successful
     info = pipeline.load()
     assert_load_info(info)
+
+
+@pytest.mark.parametrize(
+    "destination_config",
+    destinations_configs(default_sql_configs=True, default_staging_configs=True, subset=["duckdb"]),
+    ids=lambda x: x.name,
+)
+@pytest.mark.parametrize("is_none", [True, False])
+def test_warning_from_arrow_normalizer_on_null_column(
+    destination_config: DestinationTestConfiguration,
+    mocker: MockerFixture,
+    is_none: bool,
+) -> None:
+    """
+    Test that the arrow normalizer emits a warning when a pyarrow table is yielded
+    with a column (`col1`) that contains only null values.
+    """
+
+    @dlt.source()
+    def my_source():
+        @dlt.resource
+        def my_resource():
+            col1: list[Optional[str]] = [None, None] if is_none else ["a", "b"]
+
+            table = pa.table(
+                {
+                    "id": pa.array([1, 2]),
+                    "col1": pa.array(col1),
+                }
+            )
+
+            yield table
+
+        return [my_resource()]
+
+    logger_spy = mocker.spy(logger, "warning")
+
+    pipeline = destination_config.setup_pipeline("arrow_" + uniq_id())
+
+    pipeline.extract(my_source())
+    pipeline.normalize()
+
+    if is_none:
+        logger_spy.assert_called_once()
+        expected_warning = (
+            "The column col1 in table my_resource did not receive any data during this load."
```

Comment:
this is good but maybe too good :) I'd just check a substring. small wording changes should not fail this test. or? up to you